typed-store/src/rocks/mod.rs

```diff
@@ -282,6 +282,21 @@ where
         Ok(())
     }
 
+    fn clear(&self) -> Result<(), TypedStoreError> {
+        let _ = self.rocksdb.drop_cf(&self.cf);
+        self.rocksdb
+            .create_cf(self.cf.clone(), &rocksdb::Options::default())?;
+        Ok(())
+    }
+
+    fn len(&self) -> Result<usize, Self::Error> {
+        Ok(self.iter().count())
+    }
+
+    fn is_empty(&self) -> Result<bool, Self::Error> {
+        Ok(self.iter().count() == 0)
+    }
+
```

Comment:
I'd be totally favor of this if we had a way to implement these functions in O(1), instead of O(n). Since we don't these are essentially proxies for the iterator's methods.

Except the iterator's methods:
-  actually are shorter 
- do not imply having to deal with a `Result` which purpose is unclear to me at the moment,
- convey the algorithmic runtime of the function.

Reply:
Hidden O(n) would make me unhappy as a user.
Will remove these and open an issue for designing a constant time soln